Improved usage of $.Deferred and promises

ThomasBurleson opened the pull request:
Essential motivation for these changes was the _goal_ to cache only the read-only promise instances instead of the R/W Deferred instances. Source code changes include:

  1) reduced usage of variables, 
  2) improved use of $.Deferred() constructor, and 
  3) consistent use of $.when.apply( null, <promises>).

rwaldron commented:
Thomas, the changes here work great and I want to pull them in, but there are a number of "style" related issues I need you to address first... Generally I include a `readme.md` with a link to my own style guide, but didn't expect anyone to contribute so soon. The style guide is based on jQuery Core Style Guide, Google Developer Guide and Crockford's JavaScript Guide - you can review it here: https://github.com/rwldrn/idiomatic.js/blob/master/readme.md

Overall notes...
- All of the tabs have been converted to spaces; please restore to tabs.
- Don't edit `dist/adventure.js`, this file is automatically generated by the build script (I'll add notes to the readme about getting that setup)

ThomasBurleson commented:
Restored tabs and removed changes to dist/adventure.js

rwaldron commented:
I tried merging this, but what used to be 1 tab is now 2 tabs... it will require me too much work to fix formatting. Can you create a new branch, that has just the changes?

ThomasBurleson committed:
Improved use of $.Deferred and promises

Essential driver for these changes was to cache only the READ-only promise instances instead of the R/W Deferred instances.
Changes include:
- Reduced usage of variables,
- improved use of $.Deferred() constructor, and
- consistent use of $.when.apply( null, <promises>).

Signed-off-by: Thomas Burleson <ThomasBurleson@Gmail.com>

ThomasBurleson commented:
Not sure what you mean by "is now 2 tabs..."

I did a rollback to your latest change set. Then applied my changes to adventure.js.
Try it now.

ThomasBurleson committed:
Improved formatting: removed excess whitespaces and improved alignments
